packages/ag-charts-community/src/scene/transformable.ts

```diff
@@ -0,0 +1,127 @@
+import { Matrix } from './matrix';
+import { Node, SceneChangeDetection } from './node';
+
+type Constructor<T> = new (...args: any[]) => T;
+
+type MatrixTransformType<T> = T & {
+    updateMatrix(matrix: Matrix): void;
+};
+
+export function MatrixTransform<N extends Node>(Parent: Constructor<N>) {
+    const ParentNode = Parent as Constructor<Node>;
+
+    // Make sure we don't mixin `MatrixTransformInternal` multiple times.
+    let nextType = Parent;
+    while (nextType != null) {
+        if (nextType.name === 'MatrixTransformInternal') {
```

Comment:
if you're counting on the builtin `name` then this could fail after obfuscation

Reply:
Agreed and understood - I think lets proceed and see what comes out in the wash.